### ISSUE TITLE
Add severity and message to property-*-white/blacklist rules

### ISSUE THREAD

**Issue by woble** [stylelint/stylelint#1364]
> Feature request.

Allow custom severity and message for property-*-white/blacklist rules.

> Reason.

`Unexpected value "X" for property "Y"`doesn't really explain what value is accepted. One would have to dig through internal rules and guidelines to figure out what is accepted.

> Which rule, if any, is this issue related to?

For example `property-value-whitelist`

> Basic CSS example.

```css
:root {
    --order_0: 0;
}

.foo {
    z-index: 0; /* would not validate with following setup */
}

.baz {
    z-index: var(--order_0); /* would validate with following setup */
}
```

> Proposed setup.

```json
{
    "rules": {
        "property-value-whitelist": {
            "z-index" : [
                ["/^var\\(.*\\)/"],
                {
                    "severity": "error",
                    "message": "Expected a variable for property z-index."
                }
            ]
        }
    }
}
```

**Comment by evilebottnawi** [comment]
@woble https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md

**Comment by woble** [comment]
@evilebottnawi, I am aware of custom messages for rules, but I can't seem to find anything in regards to defining custom message and/or severity for a property in array within a rule. Perhaps I am overlooking something?

**Comment by evilebottnawi** [comment]
@woble may be you mean https://github.com/stylelint/stylelint/pull/865?

**Comment by woble** [comment]
I suppose #865 is more about actual formatting of the messages. I don't see how it solves what I am proposing. I might be wrong though.

Let's say I have defined the `property-value-whitelist` rule as follows, how would I go about setting a custom message for `z-index`, `background-color` and/or `color` individually.

```json
{
    "rules": {
        "property-value-whitelist": {
            "z-index": ["/^var\\(.*\\)/"],
            "background-color": ["white"],
            "color": ["black"]
        }
    }
}
```

**Comment by ntwb** [comment]
This seems a valid request to me, I've just dug around trying to determine if it's a bug or an enhancement, I think it's a bug so let's go with that 😏

**Comment by ntwb** [comment]
Just adding that example documention should be added for when using object confiuration in the docs:
• [/docs/user-guide/configuration.md#severities-error--warning](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#severities-error--warning)
• [/docs/user-guide/configuration.md#custom-messages](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#custom-messages)

**Comment by davidtheclark** [comment]
Definitely not a bug, @ntwb :) It's a very significant feature request ... And I'm not sure how feasible this is yet, so I'm labeling it "discussion".

While I'm open to reviewing a PR implementing this idea, I don't think it's a high priority for the core team. Actually, @woble, I think this might be a really good case for a plugin. And if the plugin works and is stable, we'd consider pulling it into core.

If you or someone else does work on this, I'd suggest trying to augment the configuration in a way that does not force a breaking change. For example, could the rule accept an array still, but that array takes either strings, which are treated just like now, or objects, which have the special powers you're looking for?

**Comment by jeddy3** [comment]
> Definitely not a bug, @ntwb :) It's a very significant feature request ...

Agreed.

>  I don't think it's a high priority for the core team.  Actually, @woble, I think this might be a really good case for a plugin.

Agreed.

I think only *one* severity *per rule* is needed to keep things like maintaining the formatters manageable. @woble If you want a different severity consider breaking this rule into separate plugins for your needs e.g. `declaration-z-index-var: ["always", { "severity": "warning", "message": "Use a variable for z-index property within a declaration" }]`.

Labeling as "plugin" and closing.
